examples/wallet/app/components/SendTransaction.js

```diff
@@ -1,40 +1,112 @@
   };
   const [destinationAddress, setDestinationAddress] = useState<string>('');
+  const [validAddress, setValidAddress] = useState<boolean>(false);
   const [amount, setAmount] = useState<?number>();
+  const isValidAmount = (value, currentBalance) =>
+    transactionFee + value <= currentBalance &&
+    (typeof value === 'number' && value > 0);
+  const transactionFee = feeCalculator(nodeSettings).sendFundsFee();
+  // This is asyncronous and might be resource intensive, so it's better to debounce it.
+  const debouncedAddressValidator = debounce(
+    value =>
+      isValidAddress(value)
+        .then(setValidAddress)
+        .catch(console.error),
+    config.get('formDebounceInterval')
+  );
 
   return (
-    <form onSubmit={handleSubmit}>
-      <label htmlFor="recipient">
-        Recipient:
-        <input
+    <Form onSubmit={handleSubmit}>
+      <Form.Group>
+        <Form.Label htmlFor="recipient">Recipient:</Form.Label>
+        <Form.Control
           type="text"
           name="recipient"
+          isInvalid={!validAddress && destinationAddress}
+          isValid={destinationAddress && validAddress}
           value={destinationAddress}
-          onChange={event => setDestinationAddress(event.target.value)}
-        />
-      </label>
-      <label htmlFor="amount">
-        Amount:
-        <input
-          type="number"
-          name="amount"
-          value={amount}
-          onChange={event => setAmount(event.target.value)}
+          onChange={event => {
+            debouncedAddressValidator(event.target.value);
+            setDestinationAddress(event.target.value);
+          }}
         />
-      </label>
-      <input type="submit" value="Send!" />
-    </form>
+        <Form.Control.Feedback type="invalid">
+          Please enter a valid address
+        </Form.Control.Feedback>
+      </Form.Group>
+
+      <Form.Group htmlFor="amount">
+        <Form.Label>Amount:</Form.Label>
+        <Row>
+          <Col xs="10">
+            <Form.Control
+              type="number"
+              name="amount"
+              value={amount}
+              // why the typeof checks? because we want:
+              // * The form to start with an empty value (not with zero).
+              // * Have zero be an invalid amount (this could be argued).
+              // * The form to start in a valid state.
+              isInvalid={
+                typeof amount === 'number' && !isValidAmount(amount, balance)
+              }
+              isValid={
+                typeof amount === 'number' && isValidAmount(amount, balance)
+              }
+              onChange={({ target: { value } }) => setAmount(Number(value))}
+            />
+            <Form.Control.Feedback type="invalid">
+              {typeof amount === 'number' && amount <= 0
+                ? 'Balance must be positive'
+                : `insufficient balance. Max: ${balance - transactionFee}`}
```

Comment:
Personally, I would prefer something like *Insufficient funds* or *Not enough funds*